### PR TITLE
New package: luv-1.30.1.0

### PR DESCRIPTION
**Pull request by andry-dev**
This PR adds libluv (a Lua wrapper for libuv).
The reasoning for having this is to prepare for Neovim 0.4: it requires this library to build properly.

### CODE REVIEW

File: srcpkgs/libluv-devel [hunk 1]
```diff
@@ -0,0 +1 @@
+libluv/
```
Comment (Chocimier) [inline srcpkgs/libluv-devel]: remove slash

File: srcpkgs/libluv/files/libluv.pc.in [hunk 2]
```diff
@@ -0,0 +1,12 @@
+prefix=@prefix@
```
Comment (Chocimier) [inline srcpkgs/libluv/files/libluv.pc.in]: remove file from tree, add address into distfiles

File: srcpkgs/libluv/template [hunk 3]
```diff
@@ -0,0 +1,36 @@
+# Template file for 'libluv'
+pkgname=libluv
+version=1.30.1
+revision=1
+_version_append=0
+_complete_version=${version}-${_version_append}
```
Comment (Chocimier) [inline srcpkgs/libluv/template]: please do it like in `libcli`

File: srcpkgs/libluv/template [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+archs="i686 x86_64"
```
Comment (Chocimier) [inline srcpkgs/libluv/template]: why? it builds for other archs

File: srcpkgs/libluv/template [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+wrksrc=luv-${_complete_version}
+build_style=cmake
+configure_args="-DLUA_BUILD_TYPE=System -DBUILD_MODULE=OFF -DBUILD_SHARED_LIBS=ON"
+makedepends="libuv-devel LuaJIT-devel"
+depends="lua libuv LuaJIT"
+short_desc="Bare libuv bindings for lua"
+maintainer="andry-dev <peketribal2@gmail.com>"
+license="Apache-2.0"
+homepage="https://github.com/luvit/luv"
+distfiles="https://github.com/luvit/luv/releases/download/${_complete_version}/luv-${_complete_version}.tar.gz"
```
Comment (Chocimier) [inline srcpkgs/libluv/template]: luv is better pkgname then

File: srcpkgs/libluv/template [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+checksum=420fc299f3b25aff55b839e9fe590982f45045b3e53ebeab2449eca7e23a106a
+shlib_provides="libluv.so.1"
```
Comment (Chocimier) [inline srcpkgs/libluv/template]: not needed, shlibs entry is enough

File: srcpkgs/libluv/template [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+post_extract() {
+	cp ${FILESDIR}/libluv.pc.in ${wrksrc}/libluv.pc.in
+}
+
+libluv-devel_package() {
+	depends="${makedepends} libluv>=${version}_${revision}"
+	short_desc+=" - development files"
+	pkg_install() {
+		vmove usr/include
```
Comment (Chocimier) [inline srcpkgs/libluv/template]: also .so symlink, see other templates

**Comment by andry-dev** [comment]
Thank you for the feedback

I fixed most of the stuff, there's just a small problem: I put that hack with the `libluv.pc.in` file there because the [tagged tar.gz release](https://github.com/luvit/luv/releases/download/1.30.1-0/luv-1.30.1-0.tar.gz) and the [tagged source code archive](https://github.com/luvit/luv/archive/1.30.1-0.tar.gz) differ
The former archive (which is what I used in the template) contains everything to allow proper builds _except_ the pkg-config file -- which is unfortunately required for a system install, I don't know why upstream doesn't include it
The latter archive doesn't init the submodules (so they are just empty folders, which makes the build fail), trying to call `git submodule update --init` obviously doesn't work since the archive lacks a .git folder
I tried to add the single file to `distfiles`, like so:
```
distfiles="https://github.com/luvit/luv/releases/download/${_distver}/luv-${_distver}.tar.gz 
               https://raw.githubusercontent.com/luvit/luv/${_distver}/libluv.pc.in"
```
But then xbps-src would not accept it because the file is not an archive (after fixing the checksums, obviously)

What is the best course of action then? Make it an actual patch? Am I missing something obvious?

File: srcpkgs/luv/template [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+	cp ${FILESDIR}/libluv.pc.in ${wrksrc}/libluv.pc.in
+}
+
+luv-devel_package() {
+	depends="${makedepends} libluv>=${version}_${revision}"
+	short_desc+=" - development files"
+	pkg_install() {
+		vmove usr/include
+		vmove usr/lib/pkgconfig
+		vmove usr/lib/*.so
```
Comment (Chocimier) [inline srcpkgs/luv/template]: quote

File: srcpkgs/luv/template [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+wrksrc=luv-${_distver}
+build_style=cmake
+configure_args="-DLUA_BUILD_TYPE=System -DBUILD_MODULE=OFF -DBUILD_SHARED_LIBS=ON"
+makedepends="libuv-devel LuaJIT-devel"
+depends="lua libuv LuaJIT"
+short_desc="Bare libuv bindings for lua"
+maintainer="andry-dev <peketribal2@gmail.com>"
+license="Apache-2.0"
+homepage="https://github.com/luvit/luv"
+distfiles="https://github.com/luvit/luv/releases/download/${_distver}/luv-${_distver}.tar.gz https://raw.githubusercontent.com/luvit/luv/${_distver}/libluv.pc.in"
```
Comment (Chocimier) [inline srcpkgs/luv/template]: one file per line, same for checksums

File: srcpkgs/luv/template [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+if [ "$CROSS_BUILD" -a "$XBPS_MACHINE" = "x86_64" ]; then
+	hostmakedepends+=" gcc-multilib"
+fi
+
+post_extract() {
+	cp ${XBPS_SRCDISTDIR}/luv-${version}/libluv.pc.in ${wrksrc}/libluv.pc.in
+}
+
+luv-devel_package() {
+	depends="${makedepends} libluv>=${version}_${revision}"
```
Comment (Chocimier) [inline srcpkgs/luv/template]: leftover libluv

**Comment by andry-dev** [comment]
I __think__ it should be possible to build it as a module, I have to check the CMakeLists.txt again (the last time I did that was in August, so I don't remember every option; it's kinda complicated if you take a look at it)
Now I'll fix the two comments and investigate the CMakeLists

**Comment by Chocimier** [comment]
Ok, if adding module is problematic we can go with just library under libluv name.

Side note: on new comments mails are sent, on edits they are not, so people involved into conversation do not know what happens with PR.

**Comment by andry-dev** [comment]
>Side note: on new comments mails are sent, on edits they are not, so people involved into conversation do not know what happens with PR.

Sorry, I didn't know that, I thought you would get a notification

Thank you for your patience

**Pull request merged**